### PR TITLE
Fix support for EL8

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

**Comment by treydock** [comment]
EL8 uses System apache and this fixes our assumption that apache is always SCL apache.

**Comment by treydock** [comment]
I added commit to change logic in determining SCL vs non-SCL. Doing it based on presence of SCL directory seemed like the wrong way. Basing it off contents of /etc/os-release (environment variables) would allow for future updates to support OpenSUSE and Debian based systems

File: ood-portal-generator/bin/generate [hunk 1]
```diff
@@ -5,6 +5,8 @@ SCL_SOURCE=$(command -v scl_source)
 
 ROOT_DIR="$(dirname "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")")"
 
+[ -f /etc/os-release ] && source /etc/os-release
```
Comment (treydock) [inline ood-portal-generator/bin/generate]: This isn't making it into the environment for some reason. The source for SCL is in ENV when I dump it but not the contents of /etc/os-release.

Comment (treydock) [reply ood-portal-generator/bin/generate]: I think because the environment variables in `/etc/os-release` don't use `export`, they are not accessible except in script that sourced them

**Comment by treydock** [comment]
Updated to use dotenv to load `/etc/os-release` into `ENV`.

File: ood-portal-generator/lib/ood_portal_generator/view.rb [hunk 2]
```diff
@@ -23,11 +24,18 @@ def initialize(opts = {})
       @map_fail_uri     = opts.fetch(:map_fail_uri, nil)
       @pun_stage_cmd    = opts.fetch(:pun_stage_cmd, "sudo /opt/ood/nginx_stage/sbin/nginx_stage")
 
+      Dotenv.load('/etc/os-release')
+      if (ENV['ID'] == 'rhel' || ENV['ID_LIKE'] =~ /rhel/) && ENV['VERSION_ID'] =~ /^8/
```
Comment (ericfranz) [inline ood-portal-generator/lib/ood_portal_generator/view.rb]: Dotenv makes it really clean and pithy here. 
That said, if you want to omit the need for the dependency you could replicate the version check used in the update_ood_portal bash script.

```ruby
`[ -f /etc/os-release ] && source /etc/os-release && [[ "$ID $ID_LIKE" = *"rhel"* ]] && [[ "$VERSION_ID" = "8"* ]]`
if $?.success?
  default_htpasswd = "/etc/httpd/.htpasswd"
else
  default_htpasswd = "/opt/rh/httpd24/root/etc/httpd/.htpasswd"
end
```

File: ood-portal-generator/Gemfile [hunk 3]
```diff
@@ -0,0 +1 @@
+gem 'dotenv', '~> 2.1'
```
[the remark this answers is not in the document]
Comment (treydock) [reply ood-portal-generator/Gemfile]: Change made.

**Comment by treydock** [comment]
Added commit to fix builds where somehow the build ID files generated would conflict for multiple versions of `ondemand-gems-$VERSION` so I just turned them off

**Comment by ericfranz** [comment]
@treydock is this ready to merge?

File: .travis.yml [hunk 4]
```diff
@@ -21,7 +21,7 @@ script:
       -w "/data" \
       ${OS_TYPE}:${OS_VERSION} \
       ./tests.sh
-  - shellcheck ood-portal-generator/sbin/update_ood_portal
+  - shellcheck -x ood-portal-generator/sbin/update_ood_portal
```
Comment (ericfranz) [inline .travis.yml]: after getting rid of the Gemfile etc. should we revert the change for adding `-x` to this shellcheck, or is this appropriate?

Comment (treydock) [reply .travis.yml]: It's still needed I think because other files are sourced by update_ood_portal.

Comment (treydock) [reply .travis.yml]: Yep, still needed:

```
$ docker run --rm -v "$PWD:/mnt" koalaman/shellcheck:stable ood-portal-generator/sbin/update_ood_portal 

In ood-portal-generator/sbin/update_ood_portal line 9:
[ -f /etc/os-release ] && source /etc/os-release
                                 ^-------------^ SC1091: Not following: /etc/os-release was not specified as input (see shellcheck -x).

For more information:
  https://www.shellcheck.net/wiki/SC1091 -- Not following: /etc/os-release wa...
```

**Comment by treydock** [comment]
This is ready to merge.
